src/smac_planner_hybrid.cpp

```diff
@@ -165,6 +167,116 @@ void SmacPlannerHybrid::reconfigureCB(SmacPlannerHybridConfig& config, uint32_t
 }
 
 uint32_t SmacPlannerHybrid::makePlan(
+  const geometry_msgs::PoseStamped & start,
+  const geometry_msgs::PoseStamped & goal,
+  double tolerance,
+  std::vector<geometry_msgs::PoseStamped> & plan,
+  double &cost,
+  std::string &message)
+{
+  _planning_canceled = false;
+  std::vector<geometry_msgs::PoseStamped> goal_align_poses;
+
+  if (_search_info.goal_align_distance > 0.0) {
+    geometry_msgs::PoseStamped align_pose_front, align_pose_back;
+    align_pose_front.pose = Utils::getPoseAtDistanceAlongHeading(goal.pose, -_search_info.goal_align_distance);
+    align_pose_back.pose  = Utils::getPoseAtDistanceAlongHeading(goal.pose,  _search_info.goal_align_distance);
+
+    if (!_config.allow_goal_overshoot) {
+      _search_info.setSearchBound(goal.pose);
+      _search_info.setStart(start.pose.position);
+
+      if (_search_info.isStartBehindSearchBounds()) {
+        ROS_INFO("Robot will align %f meters before the goal pose ...", _config.goal_align_distance);
+        goal_align_poses.push_back(align_pose_front);
+      } else {
+        ROS_INFO("Robot will align %f meters after the goal pose ...", _config.goal_align_distance);
+        goal_align_poses.push_back(align_pose_back);
+      }
+    } else {
+      ROS_INFO("Robot may align either %f meters before or after the goal pose...", _config.goal_align_distance);
+      goal_align_poses = {align_pose_front, align_pose_back};
+    }
+  }
+
+  // If no goal align poses, proceed with normal planning
+  if (goal_align_poses.empty()) {
+    return makeDirectPlan(start, goal, tolerance, plan, cost, message);
+  }
```

Comment:
S: move to the top and change to 
```suggestion
  if (_search_info.goal_align_distance <= 0.0) {
    return makeDirectPlan(start, goal, tolerance, plan, cost, message);
  }
```